src/shared/components/common/input-dropdown/index.tsx

```diff
@@ -0,0 +1,157 @@
+  const [isDirectInput, setIsDirectInput] = useState(false);
+  const [currentPlaceholder, setCurrentPlaceholder] = useState(placeholder);
+  const directInputRef = useRef<HTMLInputElement>(null);
+
+  const isDirectInputSelected = isDirectInput && inputValue === '직접입력';
+
+  // 직접입력 모드가 활성화되면 input에 focus
+  useEffect(() => {
+    if (isDirectInput && directInputRef.current) {
+      directInputRef.current.focus();
+    }
+  }, [isDirectInput]);
+
+  const handleSelect = (value: string) => {
+    setInputValue(value);
+    setIsDirectInput(false);
+    setCurrentPlaceholder(placeholder);
+    onChange?.(value);
+  };
+
+  const handleDirectInput = () => {
+    setInputValue('직접입력');
+    setIsDirectInput(true);
+    setCurrentPlaceholder('직접입력');
+    onChange?.(inputValue);
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setInputValue(value);
+    onChange?.(value);
+  };
+
+  const handleDirectInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setDirectInputValue(value);
+    onChange?.(value);
+  };
+
+  const selectInput = (
+    <input
+      readOnly
+      className={INPUT_STYLE + ' cursor-pointer'}
+      placeholder={currentPlaceholder}
+      type="text"
+      value={inputValue}
+      onChange={handleInputChange}
+    />
+  );
+
+  const directInput = (
+    <input
+      ref={directInputRef}
+      className={INPUT_STYLE + ' mt-2'}
+      placeholder="직접입력"
+      type="text"
+      value={directInputValue}
+      onChange={handleDirectInputChange}
+    />
+  );
+
+  return (
+    <>
+      <Dropdown
+        trigger={
+          <div className="relative">
+            {selectInput}
+            <Image
+              alt="arrow-down"
+              className="absolute top-1/2 right-24 -translate-y-1/2 lg:right-32 lg:h-36 lg:w-36"
+              height={24}
+              loading="lazy"
+              src="/icons/drop-menu-down.svg"
+              width={24}
+            />
+          </div>
+        }
+      >
+        <ul>
+          {options.map(option => (
+            <li key={option.value}>
+              <button
+                className={
+                  BTN_STYLE +
+                  (option.value === inputValue && !isDirectInput
+                    ? ' bg-mint-50/50 text-mint-300'
+                    : '')
+                }
+                onClick={() => handleSelect(option.value)}
+              >
+                {option.value}
+              </button>
+            </li>
+          ))}
+          <li>
+            <button
+              className={
+                BTN_STYLE +
+                (isDirectInputSelected ? ' bg-mint-50/50 text-mint-300' : '')
+              }
+              onClick={handleDirectInput}
+            >
+              직접입력
+            </button>
+          </li>
+        </ul>
+      </Dropdown>
+      <div className="mt-12">{isDirectInput && directInput}</div>
```

Comment:
아래와 같은 조건문보다 애초에 Dom에 렌더링 안시키는 게 더 좋을 것 같습니다.
```suggestion
    {isDirectInput &&  <div className="mt-12">{directInput}</div>}
```